notebooks/circuits_with_bloqade.py

```diff
@@ -0,0 +1,763 @@
+# %%
+from bloqade.pyqrack import StackMemorySimulator, DynamicMemorySimulator
+# StackMemorySimulator - static number of qubits
+# DynamicMemorySimulator - dynamic number of qubits, but slower. Use if you don't know how many qubits you need in advance.
+emulator = StackMemorySimulator(min_qubits=8)
+
+task = emulator.task(GHZ_state_factory, args=(4,))
+results = task.run()
+# The results is the same ResultType as the kernel return.
+# In this case, it is a list of qubits.
+print(results)
+
+
+# %% [markdown]
+# Note that, while it is instinctive to simply call `GHZ_state_factory(4)` and expect it to run, this isn't necessarily the correct abstraction. `emulator.task` links the kernel to a specific interpreter-- for example, if you wanted to run your program on a noisy emulator vs a perfect emulator vs. real hardware, this is the way you would specify it. Furthermore, the _instantiation_ of the task is not necessarily linked with the _execution_ of that task. For example, if that task must be run asynchronously on hardware, there must be an object which represents the run itself as well as the (future) results.
+# 
+# For simpler kernels that only use kirin builtin statements -- such as control flow, for loops, arithmetic, and the like -- it is possible to directly call the kernel and use the default Kirin interpreter.
+
+# %%
+try:
+    del foo
+except:
+    pass
+def foo(x:int,y:int) -> bool:
+    return x<y
+assert foo(1,2)==True
+
+# %%
+
+
+# %% [markdown]
+# ### Quantum states
+# 
+# PyQRack is mainly intended as an *emulator*, attempting to recreate the physical action and interaction of quantum hardware with classical controls. However, it is often useful for user analysis and debugging to extract out non-physical values, such as the quantum statevector, fidelity, or exact expectation values of observables and correlation functions.
+# 
+# These nonphysical values are based in being able to directly observe and evaluate the quantum state. The state can be extracted via the `PyQrackSimulator.quantum_state` method, with an input signature of a list of (pyqrack) qubits. These qubits can be generated as a return value from a kernel (as is the case for the `GHZ_state_factory` function) or from the `task.qubits()` method.
+# 
+# The resulting state is a reduced density matrix represented by its eigensystem of eigenvalues and eigenvectors. The state vector basis is the Z basis, ordered by the labels of the input qubit list with an edannes consistent with cirq. The RDM form is chosen because the qubit list may be an entangled subset of the entire register, and thus may be a mixed state. The eigensystem form is chosen because if the state is pure -- or only slightly entangled -- the resulting state will be efficiently represented by only carrying the nonzero eigenvectors of the density matrix. Consistent with `np.linalg.eigh`, the first index indexes the eigenvector (e.g. `eigenvector[:,3]` is the 4th eigenvector with eigenvalue `eigenvalue[3]`)
```

Comment:
Can you rephrase this? The use of endianness here is a bit confusing and doesn't really explain the qubit order.